pyodide_build/recipe/builder.py

```diff
@@ -798,14 +808,22 @@ def needs_rebuild(
     src_metadata
         The source section from meta.yaml.
     """
-    packaged_token = buildpath / ".packaged"
-    if not packaged_token.is_file():
-        logger.debug(
-            "%s needs rebuild because %s does not exist", pkg_root, packaged_token
+    dist_dir = pkg_root / "dist"
+    if is_wheel:
+        previous_wheel = find_matching_wheel(
+            dist_dir.glob("*.whl"), pyodide_tags(), version=version
         )
-        return True
-
-    package_time = packaged_token.stat().st_mtime
+        if not previous_wheel:
+            return True
+        package_time = previous_wheel.stat().st_mtime
```

Comment:
For wheels use the wheel itself to determine the modification time for deciding if a rebuild is needed.